File: java/log-levels/LogLevels.java
public class LogLevels {

    /**
     * Return the message part of a logLine.
     * 
     * @param logLine
     * @return String
     */
    public static String message(String logLine) {

        String[] parts = logLine.split(":");
        String result = parts[1];
        return result.strip();
    }

    /**
     * Return the logLevel part of a logLine.
     * 
     * @param logLine
     * @return String
     */
    public static String logLevel(String logLine) {

        String[] parts = logLine.split(":");
        String result = parts[0];
        result = result.strip();
        result = result.substring(1, result.length() - 1);
        return result.toLowerCase();
    }

    /**
     * Reformat a logLine to the format "msg (logLevel)"
     * 
     * @param logLine
     * @return String
     */
    public static String reformat(String logLine) {
        String msg = message(logLine);
        String lvl = logLevel(logLine);
        return msg + " (" + lvl + ")";
    }
}

File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine {
    public static string Message(string logLine) {

        string[] parts = logLine.Split(":");
        return parts[1].Trim();
    }

    public static string LogLevel(string logLine) {

        string[] parts = logLine.Split(":");
        string result = parts[0];
        result = result.Replace("[", " ");
        result = result.Replace("]", " ");
        return result.Trim().ToLower();
    }

    public static string Reformat(string logLine) {

        string msg = Message(logLine);
        string lvl = LogLevel(logLine);
        return msg + " (" + lvl + ")";
    }
}
